packages/runtime/src/runtime/modal.tsx

```diff
@@ -219,6 +246,18 @@ export const ModalWrapper: React.FC<PropsWithChildren> = ({ children }) => {
     cursor: "pointer",
   };
 
+  const getFullHeight = (options?: ModalProps): string =>
```

Comment:
What is the problem you are trying to solve for here? This seems very complex, and complex code is hard to maintain.

Reply:
there was hard-coded height being used in the code, I'm trying to get rid of that and set the height to 100 percent if it's given by the user

Reply:
We need to decide which element is responsible for styling the sizing of the modal 
![image](https://github.com/EnsembleUI/ensemble-react/assets/4132143/0b5a27fa-a619-4069-a150-4643b5c29bae)

Seems like the parent element should receive the styling props and all inner elements should size itself to that. What do youthink?

Reply:
you are right but there is one caveat in this, we'd have to give body some height in one use case when `navigateModalScreen` is used and such screen is opened inside modal which has a sticky footer. For example, this one https://github.com/EnsembleUI/ensemble-react/blob/67f44d3dec369dede5c393f27ab1f9dee3ab9605/apps/kitchen-sink/src/ensemble/screens/actions.yaml#L113-L114

Reply:
The body is always sized to parent element, and the footer is rendered over the top of the body. What would be the issue there?

Reply:
so, I'm applying full height on `.ant-modal-body` class for 2 occasions:
1. when we have full screen (i.e. `hideFullScreenIcon` is passed in as false and user taps on its icon to fully open the modal)
2. when user passes in full height using `styles` or `options` prop
For calculation, I'm just subtracting title bar height and given paddings to make the modal body of exact height that it should be.

Reply:
You should be able to achieve the same effect with minimal calculations. Consider the following structure:

```
parent div // responsible for height/width. creates a stacking context
  title/header div // responsible for containing header, can be fixed to top of parent stacking context OR inlined with content
  content div // contains body widget, sizes dynamically based on remaining space
  footer div // contains footer, fixed to bottom of parent
```

Given this structure, you should only need to calculate the offset of the content top margin based on header height.